reputation/reputation.go

```diff
@@ -0,0 +1,576 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package reputation
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/zeebo/errs"
+	// import of sqlite3 for side effects
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// StartDBError is an error class for errors related to the reputation package
+var StartDBError = errs.Class("reputation start sqlite3 error")
+
+// CreateTableError is an error class for errors related to the reputation package
+var CreateTableError = errs.Class("reputation table creation error")
+
+// InsertError is an error class for errors related to the reputation package
+var InsertError = errs.Class("reputation insertion error")
+
+// SelectError is an error class for errors related to the reputation package
+var SelectError = errs.Class("reputation selection error")
+
+// IterError is an error class for errors related to the reputation package
+var IterError = errs.Class("reputation iteration error")
+
+// DeleteError is an error class for errors related to the reputation package
+var DeleteError = errs.Class("reputation deletion error")
+
+// NodeReputationRecord is the Data type for Rows in Reputation table
+type NodeReputationRecord struct {
+	name               string
+	timestamp          string
+	uptime             int
+	auditSuccess       int
+	auditFail          int
+	latency            int
+	amountOfDataStored int
+	falseClaims        int
+	shardsModified     int
+}
+
+// startDB starts a sqlite3 database from the file path parameter
+func startDB(filePath string) (*sql.DB, error) {
+	db, err := sql.Open("sqlite3", filePath)
+	if err != nil {
+		log.Printf("%q\n", err)
```

Comment:
if you're returning an error you don't need to log it. you can let the caller log the error. this is good especially with zeebo/errs since it captures a stack trace for you